### PR TITLE
Added exception handling #2238

### PR DESCRIPTION
**Pull request by Rynder1**
Added text file with all error data i found
Added try except with logging for the found exceptions

### CODE REVIEW

**Comment by Rynder1** [comment]
Logging calls will have to be changed to fit with new logging method from Jack's PR

File: errors.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Empty text file:
```
Comment (pedrooste) [inline errors.txt]: is this a log file? usually they have the .log extension

File: onko_dicom.py [hunk 2]
```diff
@@ -67,7 +68,18 @@ def open_file(self):
...
             return
 
-        self.plot_w.plot_dcm(full_path[0])
+        try:
+            logging.log(logging.INFO, "Attempting to graph/open file")
+            self.plot_w.plot_dcm(full_path[0])
+            logging.log(logging.INFO, "successfully opened graph/file")
+
+        except pydicom.errors.InvalidDicomError as e:
+            logging.log(logging.ERROR, f"InvalidDicomError, Missing Dicom Header most likely. Error:({e})")
```
Comment (pedrooste) [inline onko_dicom.py]: most likely wording isnt the best

File: onko_dicom.py [hunk 3]
```diff
@@ -67,7 +68,18 @@ def open_file(self):
...
+        except AttributeError as e:  # very common error, seems to be everything after it has header but not full file.
+            logging.log(logging.ERROR, f"AttributeError, Missing Attribute most likely. Error:({e})")
```
Comment (pedrooste) [inline onko_dicom.py]: use the lazy format otherwise pylint picks up

Comment (pedrooste) [reply onko_dicom.py]: also use err or somthing rather than e as it shows another pylint err

File: errors.txt (hunk 1 above)
Comment (Rynder1) [reply errors.txt]: Nope, file to keep track of the different errors I have been able to produce while opening a dicom file so we can try and be more specific on what is missing etc.

File: onko_dicom.py (hunk 2 above)
Comment (Rynder1) [reply onko_dicom.py]: Fair, It was mainly as i wasn't sure if other errors would give that as its hard to try and test only having certain bits. But i can change for header as i think thats a certain. Don't think its needed for this project so will remove the most likely

File: onko_dicom.py (hunk 3 above)
Comment (Rynder1) [reply onko_dicom.py]: Whats the lazy format? Also, will change it from e

File: onko_dicom.py [hunk 4]
```diff
@@ -84,10 +85,19 @@ def open_file(self):
             logger.info("open_file user canceled open operation")
             return
 
-        self.plot_w.plot_dcm(full_path[0])
-        self.close_action.setEnabled(True)
+        try:
+            logger.info("Attempting to graph/open file")
```
Comment (sjswerdloff) [inline onko_dicom.py]: Should indicate what file.  If things go wrong, what can be done to investigate?
Where do you start?

File: onko_dicom.py [hunk 5]
```diff
@@ -84,10 +85,19 @@ def open_file(self):
...
+            self.plot_w.plot_dcm(full_path[0])
+            self.close_action.setEnabled(True)
+            logger.info("successfully opened graph/file")
+
+        except pydicom.errors.InvalidDicomError as err:
+            logger.error("InvalidDicomError, Missing Dicom Header. Error:(%s)", err)
```
Comment (sjswerdloff) [inline onko_dicom.py]: Will this include the filename?

**Review by sjswerdloff — CHANGES_REQUESTED** [review]

File: onko_dicom.py (hunk 5 above)
Comment (Rynder1) [reply onko_dicom.py]: Yep, now includes file names in errors and opening

File: onko_dicom.py (hunk 4 above)
Comment (Rynder1) [reply onko_dicom.py]: Yep, now includes file names in errors and opening

**Review by sjswerdloff — DISMISSED** [review]

**Review by pedrooste — DISMISSED** [review]

**Review by shirley9625 — DISMISSED** [review]
all good

**Review by nwilde — DISMISSED** [review]

**Review by pedrooste — APPROVED** [review]

**Review by nwilde — APPROVED** [review]

**Pull request merged**